### PR TITLE
Updated the VOL class struct version for the v1.14 release of HDF5

### PR DESCRIPTION
**Pull request by raylu-hdf**

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]

**Comment by hyoklee** [comment]
You may want to update README.md as well. It has still 1.13.

**Comment by raylu-hdf** [comment]
> You may want to update README.md as well. It has still 1.13.

Thanks for pointing it out.  I updated it.

File: src/template_vol_connector.c [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 /* The VOL class struct */
 static const H5VL_class_t template_class_g = {
-    2,                                              /* VOL class struct version */
```
Comment (jhendersonHDF) [inline src/template_vol_connector.c:27]: I believe it may be more correct to update this to "3" rather than using the H5VL_VERSION macro.

**Comment by jhendersonHDF** [comment]
Note that updating this VOL will (probably) cause it to load correctly, so then we're left with a question of whether we should inform the user when we try to load a plugin but it contains too old of a VOL class struct. And if we should do so, how we should do that in a nice way.

**Review by jhendersonHDF — APPROVED** [review]

**Review by derobins — APPROVED** [review]

**Pull request merged**